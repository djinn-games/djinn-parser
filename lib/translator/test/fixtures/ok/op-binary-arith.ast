{
  "type": "Program",
  "name": {
    "type": "Identifier",
    "name": "binary_arith",
    "line": 1
  },
  "body": {
    "type": "Body",
    "sentences": [
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "OperationExpression",
          "left": {
            "type": "Literal",
            "line": 3,
            "dataType": "int",
            "value": 2
          },
          "right": {
            "type": "Literal",
            "line": 3,
            "dataType": "int",
            "value": 2
          },
          "operator": "add",
          "line": 3
        },
        "line": 3
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "OperationExpression",
          "left": {
            "type": "OperationExpression",
            "left": {
              "type": "OperationExpression",
              "right": {
                "type": "Literal",
                "line": 4,
                "dataType": "int",
                "value": 2
              },
              "operator": "minus",
              "line": 4
            },
            "right": {
              "type": "Literal",
              "line": 4,
              "dataType": "float",
              "value": 5
            },
            "operator": "sub",
            "line": 4
          },
          "right": {
            "type": "Literal",
            "line": 4,
            "dataType": "int",
            "value": 3
          },
          "operator": "sub",
          "line": 4
        },
        "line": 4
      },
      {
        "type": "ExpressionSentence",
        "expression": {
          "type": "OperationExpression",
          "left": {
            "type": "Literal",
            "line": 5,
            "dataType": "bool",
            "value": false
          },
          "right": {
            "type": "Literal",
            "line": 5,
            "dataType": "str",
            "value": ""
          },
          "operator": "add",
          "line": 5
        },
        "line": 5
      }
    ],
    "line": 2
  },
  "line": 1
}