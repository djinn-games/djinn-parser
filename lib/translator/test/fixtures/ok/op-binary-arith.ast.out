{
  "type": "Program",
  "body": [
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "BinaryExpression",
        "operator": "+",
        "left": {
          "type": "Literal",
          "value": 2,
          "raw": "2"
        },
        "right": {
          "type": "Literal",
          "value": 2,
          "raw": "2"
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "BinaryExpression",
        "operator": "-",
        "left": {
          "type": "BinaryExpression",
          "operator": "-",
          "left": {
            "type": "UnaryExpression",
            "operator": "-",
            "prefix": true,
            "argument": {
              "type": "Literal",
              "value": 2,
              "raw": "2"
            }
          },
          "right": {
            "type": "Literal",
            "value": 5,
            "raw": "5"
          }
        },
        "right": {
          "type": "Literal",
          "value": 3,
          "raw": "3"
        }
      }
    },
    {
      "type": "ExpressionStatement",
      "expression": {
        "type": "BinaryExpression",
        "operator": "+",
        "left": {
          "type": "Literal",
          "value": false,
          "raw": "false"
        },
        "right": {
          "type": "Literal",
          "value": "",
          "raw": "\"\""
        }
      }
    }
  ]
}
